.github/workflows/code-freeze.yml

```
@@ -0,0 +1,35 @@
+name: Code Freeze
+
+on:
+  pull_request:
+    branches:
+      - main
+
```

Can we add a top level permissions limit for the workflow:
```suggestion

permissions:
  contents: read
```